service/worker/scanner/shardscanner/types.go

```diff
@@ -419,6 +414,69 @@ func (s Shards) Flatten() ([]int, int, int) {
 	return shardList, min, max
 }
 
-func (c ScannerContextKey) String() string {
-	return string(c)
+// NewShardScannerContext sets scanner context up
+func NewShardScannerContext(
+	res resource.Resource,
+	config *ScannerConfig,
+) Context {
```

Comment:
I am a bit sick of this name `Context`. Could we make a more explicit? maybe ShardScannerContext. It will help people with less knowledge or you become less familar after 1 year ago, read it easier I think